### PR TITLE
Banyan Python

### PR DESCRIPTION
**Pull request by clarisw**

### CODE REVIEW

File: Banyan/src/queues.jl [hunk 1]
```diff
@@ -168,20 +167,6 @@ function send_to_client(value_id, value, worker_memory_used = 0)
...
-
-
-
-
-
-
-
-
-
-
```
Comment (cailinw) [inline Banyan/src/queues.jl:184]: whoahhhhhhhh!!!!

File: Banyan/src/sessions.jl [hunk 2]
```diff
@@ -112,6 +114,7 @@ function start_session(;
     end
     s3_bucket_name = get_cluster_s3_bucket_name(cluster_name)
 
+    # TODO: Construct this dictionary
```
Comment (cailinw) [inline Banyan/src/sessions.jl:117]: I think you can take this comment off, since this is no longer a TODO!

File: Banyan/src/sessions.jl [hunk 3]
```diff
@@ -137,6 +140,7 @@ function start_session(;
                 s3_put(get_aws_config(), s3_bucket_name, "$(environment_hash)/Manifest.toml", manifest_toml)
             end
         end
+    # TODO: Else case is the same
```
Comment (cailinw) [inline Banyan/src/sessions.jl]: I think you can take this comment off, since this is no longer a TODO!

File: Banyan/src/sessions.jl [hunk 4]
```diff
@@ -165,6 +169,7 @@ function start_session(;
         end
     end
     # TODO: Optimize so that we only upload (and download onto cluster) the files if the filename doesn't already exist
+    # Example of f might be "C:/Users/Melany Winston/.../src/clusters.py" --> "cluster.py" extracting out cluster.py
```
Comment (cailinw) [inline Banyan/src/sessions.jl]: idk if you want to remove your full name from the example here :)

File: Banyan/src/sessions.jl [hunk 5]
```diff
@@ -470,7 +476,7 @@ function run_session(;
                     directory = directory, dev_paths = dev_paths, force_sync = force_sync, force_pull = force_pull, force_install = force_install, 
                     estimate_available_memory = estimate_available_memory, nowait = false, email_when_ready = email_when_ready, for_running = true)
     catch
-        session_id = try
+        try
```
Comment (cailinw) [inline Banyan/src/sessions.jl]: I think we need to actually set `session_id` to either `get_session_id()` or `nothing.

So,

```
session_id = nothing
 try
            session_id = get_session_id()
        catch
            session_id = nothing
        end
```

**Review by cailinw — CHANGES_REQUESTED** [review]

File: Banyan/src/sessions.jl (hunk 5 above)
Comment (calebwin) [reply Banyan/src/sessions.jl]: Resolved

File: Banyan/src/sessions.jl (hunk 4 above)
Comment (calebwin) [reply Banyan/src/sessions.jl]: Resolved

File: Banyan/src/sessions.jl (hunk 3 above)
Comment (calebwin) [reply Banyan/src/sessions.jl]: Yup

File: Banyan/src/sessions.jl (hunk 2 above)
Comment (calebwin) [reply Banyan/src/sessions.jl:117]: K

File: Banyan/src/queues.jl (hunk 1 above)
Comment (calebwin) [reply Banyan/src/queues.jl:184]: :)

**Review by calebwin — APPROVED** [review]

**Pull request merged**